nixos/modules/services/networking/go-away.nix

```diff
@@ -0,0 +1,257 @@
+                  else
+                    null;
+                type = types.nullOr types.str;
+              };
+
+              tlsKey = lib.mkOption {
+                default =
+                  if config.useACMEHost != null then
+                    "${hostConfig.security.acme.certs.${config.useACMEHost}.directory}/key.pem"
+                  else
+                    null;
+                type = types.nullOr types.str;
+              };
+
+              config = lib.mkOption {
+                default = { };
+                description = ''
+                  TODO
+                '';
+                type = types.submodule {
+                  freeformType = yamlFormat.type;
+                  options = {
+                    bind = lib.mkOption {
+                      default = { };
+                      type = types.submodule {
+                        freeformType = yamlFormat.type;
+                        options = {
+                          address = lib.mkOption {
+                            type = types.str;
+                            default = "${config.bindHost}:${builtins.toString config.bindPort}";
+                            readOnly = true;
+                          };
+                          tls-certificate = lib.mkOption {
+                            type = types.nullOr types.str;
+                            default =
+                              if config.tlsCertificate != null then
+                                "/run/credentials/${instanceName name}.service/tlsCertificate"
+                              else
+                                null;
+                            readOnly = true;
+                          };
+                          tls-key = lib.mkOption {
+                            type = types.nullOr types.str;
+                            default =
+                              if config.tlsKey != null then "/run/credentials/${instanceName name}.service/tlsKey" else null;
+                            readOnly = true;
+                          };
+                          # TODO
+                        };
+                      };
+                    };
+                    # TODO
+                  };
+                };
+              };
+
+              policy = lib.mkOption {
+                default = { };
+                description = ''
+                  TODO
+                '';
+                type = types.submodule {
+                  freeformType = yamlFormat.type;
+                  options = {
+                    # TODO
+                  };
+                };
+              };
+            };
+          }
+        )
+      );
+    };
+  };
+
+  config = lib.mkIf (enabledInstances != { }) {
+    environment.etc = lib.mapAttrs' (
+      name: instance:
+      lib.nameValuePair (policyEtcPath name) ({
+        source = policyFile name instance;
+      })
+    ) enabledInstances;
```

Comment:
I don't think there's a need to write this to /etc, passing it in ExecStart should be enough. Unless of course go-aways supports service reloading with a full restart, then we can do the same thing as with the Nginx module.

Reply:
I think it's neat to expose the current config below /etc/ for browsing it.

If it supports reloads, then source file needs to be added to `reloadTriggers`, since the target symlink won't trigger reloads.